python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/ethernet-interfaces.j2

```diff
@@ -160,6 +160,103 @@ interface {{ ethernet_interface.name }}
+{%     endif %}
+{%     if ethernet_interface.switchport.tap.encapsulation.gre is arista.avd.defined %}
+{%         for gre_strip_config in ethernet_interface.switchport.tap.encapsulation.gre %}
+{%             if gre_strip_config.strip is arista.avd.defined(true) %}
+{%                 set tap_encapsulation_cli = "switchport tap encapsulation gre" %}
+{%                 if gre_strip_config.destination is arista.avd.defined %}
+{%                     set tap_encapsulation_cli = tap_encapsulation_cli ~ " destination " ~ gre_strip_config.destination %}
+{%                     if gre_strip_config.source is arista.avd.defined %}
+{%                         set tap_encapsulation_cli = tap_encapsulation_cli ~ " source " ~ gre_strip_config.source %}
+{%                     endif %}
+{%                     if gre_strip_config.protocol is arista.avd.defined %}
+{%                         set tap_encapsulation_cli = tap_encapsulation_cli ~ " protocol " ~ gre_strip_config.protocol %}
+{%                     endif %}
+{%                 elif gre_strip_config.protocol is arista.avd.defined and gre_strip_config.destination is not arista.avd.defined %}
+{%                     set tap_encapsulation_cli = tap_encapsulation_cli ~ " protocol " ~ gre_strip_config.protocol %}
+{%                 endif %}
+   {{ tap_encapsulation_cli }} strip
+{%             endif %}
+{%         endfor %}
+{%     endif %}
+{%     if ethernet_interface.switchport.tool.mpls_pop_all is arista.avd.defined(true) %}
+   switchport tool mpls pop all
+{%     endif %}
+{%     if ethernet_interface.switchport.tool.encapsulation.vn_tag_strip is arista.avd.defined(true) %}
+   switchport tool encapsulation vn-tag strip
+{%     endif %}
+{%     if ethernet_interface.switchport.tool.encapsulation.dot1br_strip is arista.avd.defined(true) %}
+   switchport tool encapsulation dot1br strip
+{%     endif %}
+{%     if ethernet_interface.switchport.tap.allowed_vlan is arista.avd.defined %}
+   switchport tap allowed vlan {{ ethernet_interface.switchport.tap.allowed_vlan }}
+{%     endif %}
+{%     if ethernet_interface.switchport.tool.allowed_vlan is arista.avd.defined %}
+   switchport tool allowed vlan {{ ethernet_interface.switchport.tool.allowed_vlan }}
+{%     endif %}
+{%     if ethernet_interface.switchport.tool.identity.tag is arista.avd.defined %}
+   switchport tool identity {{ ethernet_interface.switchport.tool.identity.tag }}
+{%     endif %}
+{%     if ethernet_interface.switchport.tool.identity.dot1q_dzgre_source is arista.avd.defined %}
+   switchport tool identity dot1q source dzgre {{ ethernet_interface.switchport.tool.identity.dot1q_dzgre_source }}
+{%     elif ethernet_interface.switchport.tool.identity.qinq_dzgre_source is arista.avd.defined %}
+   switchport tool identity qinq source dzgre {{ ethernet_interface.switchport.tool.identity.qinq_dzgre_source }}
+{%     endif %}
+{%     if ethernet_interface.switchport.tap.truncation.enabled is arista.avd.defined %}
```

Comment:
do you think we should check enabled key as arista.avd.defined(true).
Because we made it as required but user can give value as false. I did not tried but lets check

Reply:
right, it should be checked `true`. I will update.
Thanks.